dashboard/lib/services/curriculum_pdfs.rb

```diff
@@ -128,8 +129,24 @@ def self.regenerate_pdfs(scripts)
 
           script.lessons.select(&:has_lesson_plan).each do |lesson|
             puts "Regenerating Lesson PDFs for #{lesson.key} (from #{script.name})"
-            generate_lesson_pdf(lesson, dir)
+            # generate PDF and get the pathname to the generated PDF
+            pdf_pathname = generate_lesson_pdf(lesson, dir)
             any_pdf_generated = true
+
+            # Generate matadata
+            if any_pdf_generated
+              metadata ={
+                course: script.name,
+                unit_fullname: script.localized_title,
+                unit: script.unit_number ? format("U%02d", script.unit_number) : "U1",
```

Comment:
I'd love to talk a bit more about the `unit_fullname` vs `unit` fields here I had looked at [this](https://github.com/code-dot-org/code-dot-org/pull/63053) PR, but from your comments in the tech spec, perhaps there is newer work that I should look at - I couldn't find any newer PRs in github when I searched for it though.